samples/c/whisper_speech_recognition/whisper_speech_recognition.c

```diff
@@ -0,0 +1,471 @@
+    if (!*audio_data) {
+        fprintf(stderr, "Error: Cannot allocate memory for audio data\n");
+        fclose(file);
+        return -1;
+    }
+
+    // Read and convert audio data to float
+    if (header.bits_per_sample == 16) {
+        int16_t* temp_buffer = (int16_t*)malloc(num_samples * sizeof(int16_t));
+        if (!temp_buffer) {
+            fprintf(stderr, "Error: Cannot allocate temporary buffer\n");
+            free(*audio_data);
+            fclose(file);
+            return -1;
+        }
+
+        if (fread(temp_buffer, sizeof(int16_t), num_samples, file) != num_samples) {
+            fprintf(stderr, "Error: Cannot read audio data\n");
+            free(temp_buffer);
+            free(*audio_data);
+            fclose(file);
+            return -1;
+        }
+
+        // Convert 16-bit PCM to float [-1, 1]
+        for (size_t i = 0; i < num_samples; i++) {
+            (*audio_data)[i] = temp_buffer[i] / 32768.0f;
+        }
+
+        free(temp_buffer);
+    } else if (header.bits_per_sample == 32) {
+        if (fread(*audio_data, sizeof(float), num_samples, file) != num_samples) {
+            fprintf(stderr, "Error: Cannot read audio data\n");
+            free(*audio_data);
+            fclose(file);
+            return -1;
+        }
+    } else {
+        fprintf(stderr, "Error: Unsupported bit depth: %d\n", header.bits_per_sample);
+        free(*audio_data);
+        fclose(file);
+        return -1;
+    }
+
+    fclose(file);
+    return 0;
+}
+
+// Generate synthetic audio (sine wave)
+void generate_synthetic_audio(float* audio, size_t length, float frequency, float sample_rate) {
+    for (size_t i = 0; i < length; i++) {
+        audio[i] = 0.5f * sinf(2.0f * M_PI * frequency * (float)i / sample_rate);
+    }
+}
+
+// Resample audio to 16kHz if needed (simple linear interpolation)
+float* resample_audio(const float* input, size_t input_length, float input_rate, float target_rate, size_t* output_length) {
+    if (input_rate == target_rate) {
+        *output_length = input_length;
+        float* output = (float*)malloc(input_length * sizeof(float));
+        if (output) {
+            memcpy(output, input, input_length * sizeof(float));
+        }
+        return output;
+    }
+
+    float ratio = input_rate / target_rate;
+    *output_length = (size_t)(input_length / ratio);
+    float* output = (float*)malloc(*output_length * sizeof(float));
+    
+    if (!output) {
+        return NULL;
+    }
+
+    for (size_t i = 0; i < *output_length; i++) {
+        float src_idx = i * ratio;
+        size_t idx0 = (size_t)src_idx;
+        size_t idx1 = idx0 + 1;
+        
+        if (idx1 >= input_length) {
+            output[i] = input[input_length - 1];
+        } else {
+            float frac = src_idx - idx0;
+            output[i] = input[idx0] * (1.0f - frac) + input[idx1] * frac;
+        }
+    }
+
+    return output;
+}
+
+
+int main(int argc, char* argv[]) {
```

Comment:
I think it makes sense to move `parse_arguments`, `load_wav_file`, `generate_synthetic_audio`, `resample_audio` to a separate utils file.